payments-core/src/test/java/com/stripe/android/payments/core/authentication/threeds2/Stripe3ds2TransactionActivityTest.kt

```diff
@@ -75,6 +74,49 @@ class Stripe3ds2TransactionActivityTest {
         }
     }
 
+    @Test
+    fun `Stripe3ds2TransactionViewModelFactory gets initialized by Injector when Injector is available`() {
+        val injector = object : Injector {
+            override fun inject(injectable: Injectable<*>) {
+                val factory = injectable as Stripe3ds2TransactionViewModelFactory
+                factory.stripeRepository = mock()
+                factory.analyticsRequestExecutor = mock()
+                factory.analyticsRequestFactory = mock()
+                factory.messageVersionRegistry = mock()
+                factory.threeDs2Service = mock()
+                factory.challengeResultProcessor = mock()
+                factory.workContext = mock()
+            }
+        }
+        WeakMapInjectorRegistry.register(injector, INJECTOR_KEY)
+
+        ActivityScenario.launch<Stripe3ds2TransactionActivity>(
+            Intent(
+                ApplicationProvider.getApplicationContext(),
+                Stripe3ds2TransactionActivity::class.java
+            ).putExtras(
+                ARGS.toBundle()
+            )
+        ).use { activityScenario ->
+            assertThat(activityScenario.state).isEqualTo(Lifecycle.State.RESUMED)
```

Comment:
The activity will only be in `RESUMED` state if the viewmodel is correctly initialized